pkg/api/norman/store/cluster/cluster_store.go

```diff
@@ -970,39 +976,54 @@ func (r *Store) migrateSecrets(data map[string]interface{}, currentReg, currentS
 		return secrets{}, err
 	}
 	if s.regSecret != nil {
-		data[registrySecretKey] = s.regSecret.Name
+		values.PutValue(data, s.regSecret.Name, "clusterSecrets", registrySecretKey)
+		if _, ok := data[registrySecretKey]; ok {
+			delete(data, registrySecretKey) // migrate from status to spec
+		}
 		rkeConfig.PrivateRegistries = secretmigrator.CleanRegistries(rkeConfig.PrivateRegistries)
 	}
 	s.s3Secret, err = r.secretMigrator.CreateOrUpdateS3Secret(currentS3, rkeConfig, nil)
```

Comment:
It just occurred to me, but we should probably be setting the owner references on these secrets because anyone that has tried to create them and have it not persist to status just has a bunch of orphaned secrets floating around in `cattle-global-data`.

Reply:
Good call, I've added that